Fails to build on Linux 4.10

Compile time issues:
* `req_op` [was renamed](https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/patch/include/linux/blk_types.h?id=ef295ecf090d3e86e5b742fc6ab34f1122a43773) `req_opf`
* `READ_SYNC` [has been dropped](https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/patch/include/linux/fs.h?id=70fd76140a6cb63262bd47b68d57b42e889c10ee).
```shell
if [ ! -f kernel-config.h ] || tail -1 kernel-config.h | grep -qv '#endif'; then ./genconfig.sh 4.10.0-0.rc7.git1.1.fc26.x86_64; fi;
make -C /lib/modules/4.10.0-0.rc7.git1.1.fc26.x86_64/build M=/root/dattobd/src modules
make[1]: Entering directory '/usr/src/kernels/4.10.0-0.rc7.git1.1.fc26.x86_64'
  CC [M]  /root/dattobd/src/dattobd.o
/root/dattobd/src/dattobd.c:254:62: warning: ‘enum req_op’ declared inside parameter list will not be visible outside of this definition or declaration
 static inline void dattobd_set_bio_ops(struct bio *bio, enum req_op op, unsigned op_flags){
                                                              ^~~~~~
/root/dattobd/src/dattobd.c:254:69: error: parameter 2 (‘op’) has incomplete type
 static inline void dattobd_set_bio_ops(struct bio *bio, enum req_op op, unsigned op_flags){
                                                                     ^~
/root/dattobd/src/dattobd.c:254:20: error: function declaration isn’t a prototype [-Werror=strict-prototypes]
 static inline void dattobd_set_bio_ops(struct bio *bio, enum req_op op, unsigned op_flags){
                    ^~~~~~~~~~~~~~~~~~~
/root/dattobd/src/dattobd.c: In function ‘snap_handle_read_bio’:
/root/dattobd/src/dattobd.c:2443:40: error: ‘READ_SYNC’ undeclared (first use in this function); did you mean ‘REQ_SYNC’?
  dattobd_set_bio_ops(bio, REQ_OP_READ, READ_SYNC);
                                        ^~~~~~~~~
                                        REQ_SYNC
/root/dattobd/src/dattobd.c:2443:40: note: each undeclared identifier is reported only once for each function it appears in
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:301: /root/dattobd/src/dattobd.o] Error 1
make[1]: *** [Makefile:1494: _module_/root/dattobd/src] Error 2
make[1]: Leaving directory '/usr/src/kernels/4.10.0-0.rc7.git1.1.fc26.x86_64'
make: *** [Makefile:10: default] Error 2
```

--- a/src/configure-tests/feature-tests/enum_req_opf.c
+++ b/src/configure-tests/feature-tests/enum_req_opf.c
@@ -1,0 +1,16 @@
+/*
+    Copyright (C) 2017 Datto Inc.
+
+    This file is part of dattobd.
+
+    This program is free software; you can redistribute it and/or modify it
+    under the terms of the GNU General Public License version 2 as published
+    by the Free Software Foundation.
+*/
+
+#include "../../includes.h"
+
+static inline void dummy(void){
+	enum req_opf n = 0;
+	(void)n;
+}

--- a/src/dattobd.c
+++ b/src/dattobd.c
@@ -188,6 +188,10 @@ static struct block_device *blkdev_get_by_path(const char *path, fmode_t mode, v
 }
 #endif
 
+#ifndef READ_SYNC
+#define READ_SYNC 0
+#endif
+
 #ifndef REQ_WRITE
 #define REQ_WRITE WRITE
 #endif
@@ -211,16 +215,16 @@ static struct block_device *blkdev_get_by_path(const char *path, fmode_t mode, v
 #define REQ_DISCARD 0
 #endif
 
-enum req_op {
+typedef enum req_op {
 	REQ_OP_READ,
 	REQ_OP_WRITE,
 	REQ_OP_DISCARD,         /* request to discard sectors */
 	REQ_OP_SECURE_ERASE,    /* request to securely erase sectors */
 	REQ_OP_WRITE_SAME,      /* write same block many times */
 	REQ_OP_FLUSH,           /* request for cache flush */
-};
+} req_op_t;
 
-static inline void dattobd_set_bio_ops(struct bio *bio, enum req_op op, unsigned op_flags){
+static inline void dattobd_set_bio_ops(struct bio *bio, req_op_t op, unsigned op_flags){
 	bio->bi_rw = 0;
 
 	switch(op){
@@ -251,7 +255,14 @@ static inline void dattobd_set_bio_ops(struct bio *bio, enum req_op op, unsigned
 	#define dattobd_submit_bio_wait(bio) submit_bio_wait(0, bio)
 #else
 
-static inline void dattobd_set_bio_ops(struct bio *bio, enum req_op op, unsigned op_flags){
+#ifndef HAVE_ENUM_REQ_OPF
+//#if LINUX_VERSION_CODE < KERNEL_VERSION(4,10,0)
+typedef enum req_op req_op_t;
+#else
+typedef enum req_opf req_op_t;
+#endif
+
+static inline void dattobd_set_bio_ops(struct bio *bio, req_op_t op, unsigned op_flags){
 	bio->bi_opf = 0;
 	bio_set_op_attrs(bio, op, op_flags);
 }

